Neos.ContentRepository.Core/Classes/Feature/NodeCreation/Dto/NodeAggregateIdsByNodePaths.php

```diff
@@ -95,7 +97,34 @@ public static function fromJsonString(string $jsonString): self
 
     public function merge(self $other): self
     {
-        return new self(array_merge($this->nodeAggregateIds, $other->getNodeAggregateIds()));
+        return new self(array_merge($this->nodeAggregateIds, $other->nodeAggregateIds));
+    }
+
+    public function completeForNodeOfType(NodeTypeName $nodeTypeName, NodeTypeManager $nodeTypeManager): self
+    {
+        return self::fromArray($this->createNodeAggregateIdsForNodeType($nodeTypeName, $nodeTypeManager))
+            ->merge($this);
+    }
```

Comment:
i really like this method or here :D and was thinking as well to put it here.

(originally it was hidden here https://github.com/neos/neos-development-collection/blob/10e6373b5e45e6ac119a319c8d79c1e67cdcdb87/Neos.ContentRepository.Core/Classes/Feature/NodeCreation/NodeCreation.php#L344-L367)

Putting it here exposed will be important for the node templates package as we need it there.

Though i would prefer if there also was a static factory for this so that i dont have to write

`NodeAggregateIdsByNodePaths::createEmpty()->completeForNodeOfType(...)`

Reply:
like this?

Reply:
yes looks pretty practical - and one can easily find this method.

```php
$nodeAggregateIdsByNodePaths = NodeAggregateIdsByNodePaths::createForNodeType($nodeTypeName, $nodeTypeManager);
```

ideally we should enrich also the doc comments for the both methods about the _why_ one would want to do this. (And  maybe also a little about the "what" - that it recursively generates node ids for all declared tethered paths.)